tests/unit/symmetric_tree.c

```diff
@@ -0,0 +1,221 @@
+/*
+   IGraph library.
+   Copyright (C) 2021  The igraph development team <igraph@igraph.org>
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; either version 2 of the License, or
+   (at your option) any later version.
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+   
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include <igraph.h>
+#include "test_utilities.inc"
+
+#define PRINT_DESTROY(name) \
+    printf(name "\n"); \
+    print_graph_canon(&g); \
+    igraph_destroy(&g); \
+    igraph_vector_int_destroy(&v); \
+    printf("\n");
+
+int main() {
+
+    igraph_t g, g_test;
+    igraph_vector_int_t v, edges;
+    igraph_bool_t same;
+    igraph_integer_t vertices;
+
+    /**** Test with empty vector ****/
+    // initialize variables for test
+    igraph_vector_int_init_int(&v, 0);
+    // test
+    IGRAPH_ASSERT(igraph_symmetric_tree(&g, &v, IGRAPH_TREE_UNDIRECTED) == IGRAPH_SUCCESS);
+    // root vertice always gets created, cannot create null graph with this function
+    IGRAPH_ASSERT(igraph_vcount(&g) == 1);
+    IGRAPH_ASSERT(igraph_ecount(&g) == 0);
+    //clean up
+    PRINT_DESTROY("Singleton graph");
```

Comment:
You're now checking everything with assertions and the printing the graph.
Instead you can remove the assertions and check the output, by adding a .out file with the expected output. That should clean up your code a lot.